src/Common/src/TypeSystem/IL/Stubs/ArrayMethodILEmitter.cs

```diff
@@ -136,16 +135,19 @@ void EmitILForAccessor()
 
             codeStream.Emit(ILOpcode.ret);
 
-#if false
             codeStream.EmitLdc(0);
             codeStream.EmitLabel(rangeExceptionLabel1); // Assumes that there is one "int" pushed on the stack
             codeStream.Emit(ILOpcode.pop);
 
-            var tokIndexOutOfRangeCtorExcep = GetToken(GetException(kIndexOutOfRangeException).GetDefaultConstructor());
+            TypeDesc indexOutOfRangeException = _method.Context.GetWellKnownType(WellKnownType.IndexOutOfRangeException);
+            int tokIndexOutOfRangeCtorExcep = NewToken(indexOutOfRangeException.GetDefaultConstructor());
+#if false
             codeStream.EmitLabel(rangeExceptionLabel);
-            codeStream.Emit(ILOpcode.newobj, tokIndexOutOfRangeCtorExcep, 0);
+#endif
+            codeStream.Emit(ILOpcode.newobj, tokIndexOutOfRangeCtorExcep);
             codeStream.Emit(ILOpcode.throw_);
```

Comment:
It may be nice to just call ThrowOutOfRangeException helper here instead of inlining it to every expansion. 

BTW: We will need ability to call a helper to implement the type compatibility check here - it should be prepared for this.